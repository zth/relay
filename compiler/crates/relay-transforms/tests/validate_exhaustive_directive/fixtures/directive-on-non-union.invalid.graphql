#expected-to-throw
fragment DirectiveOnNonUnion on User {
  name @exhaustive
}
# %extensions%
directive @exhaustive(ignore: [String!]) on FIELD
