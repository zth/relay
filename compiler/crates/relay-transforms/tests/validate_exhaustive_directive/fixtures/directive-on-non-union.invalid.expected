==================================== INPUT ====================================
#expected-to-throw
fragment DirectiveOnNonUnion on User {
  name @exhaustive
}
# %extensions%
directive @exhaustive(ignore: [String!]) on FIELD
==================================== ERROR ====================================
✖︎ The @exhaustive directive can only be applied to fields returning union types.

  directive-on-non-union.invalid.graphql:3:3
    2 │ fragment DirectiveOnNonUnion on User {
    3 │   name @exhaustive
      │   ^^^^
    4 │ }
